Alpaca takes a long time to fail to download PAC script after disconnecting from VPN

My work stores our PAC script at a URL that's only accessible via a VPN connection. It seems like there's some issue with alpaca detecting that it can't reach the URL anymore. When I start alpaca not on the VPN I see that it tries to download the PAC script, but fails due to a failed DNS lookup:
```
2026/02/11 11:31:00.970762 pacfetcher.go:109: Error downloading PAC file, giving up: "Get \"https://REDACTED/proxy.pac\": dial tcp: lookup REDACTED: no such host"
```

This is the behavior that I'd expect to see. When I then connect to the VPN everything works fine (it fetches the PAC script, and proxies things correctly), but when I disconnect from the VPN again later it seems to need to wait for a timeout instead of seeing that the DNS lookup failed. Alpaca then also retries the download which results in a 1-2 minute delay for any queries that occur while alpaca is failing to get the PAC script

```
2026/02/11 11:31:47.285424 pacfetcher.go:100: Attempting to download PAC from https://REDACTED/proxy.pac
2026/02/11 11:32:17.314834 pacfetcher.go:105: Error downloading PAC file, will retry after 2s: "Get \"https://REDACTED/proxy.pac\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"
2026/02/11 11:32:49.317431 pacfetcher.go:109: Error downloading PAC file, giving up: "Get \"https://REDACTED/proxy.pac\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"
```

Once alpaca has given up it seems to start responding faster, but that delay after disconnecting my VPN is quite annoying
```
2026/02/11 11:40:17.693459 proxyfinder.go:97: [26] CONNECT //REDACTED:443 via "DIRECT" (not connected to PAC server)
```

Not sure if it has anything to do with it, but I'm trying this on NixOS with my VPN being configured by the networkmanager-openvpn plugin

Sorry it took me so long to get back to you (work has been weirdly busy, so I didn't have time to test this), but it looks like #166 generally fixes this issue for me. After disconnecting the VPN, it goes back to the normal very slight delay of waiting for the DNS lookup to fail before giving up on downloading the PAC file

```
2026/03/11 10:36:04.227722 pacfetcher.go:152: Attempting to download PAC from https://REDACTED/proxy.pac
2026/03/11 10:36:04.242415 pacfetcher.go:169: Error downloading PAC file, will retry after 2s: "Get \"https://REDACTED/proxy.pac\": dial tcp: lookup REDACTED: no such host"
2026/03/11 10:36:06.243695 pacfetcher.go:173: Error downloading PAC file, giving up: "Get \"https://REDACTED/proxy.pac\": dial tcp: lookup REDACTED: no such host"
```

```
❯ time curl https://REDACTED/
curl: (56) CONNECT tunnel failed, response 502

real	0m2.020s
user	0m0.003s
sys	0m0.009s

❯ time curl https://REDACTED/
curl: (56) CONNECT tunnel failed, response 502

real	0m0.020s
user	0m0.013s
sys	0m0.004s
```

I am curious if it would be possible to just cache the PAC file instead of needing to attempt to re-download it when changing networks. I tried downloading the PAC file locally and having alpaca use that (I couldn't find a way to get it to read from a local file for some reason, so I just ran `python -m http.server` and pointed it to that), and then there was no delay after disconnecting my VPN

```
# VPN connected
2026/03/11 10:49:09.084981 proxyfinder.go:113: [3] CONNECT //REDACTED:443 via "DIRECT"
# VPN disconnected
2026/03/11 10:49:14.710732 pacfetcher.go:152: Attempting to download PAC from http://localhost:8000/proxy.pac
2026/03/11 10:49:14.715664 proxyfinder.go:113: [4] CONNECT //REDACTED:443 via "DIRECT"
2026/03/11 10:49:14.738252 proxy.go:142: [4] Error dialling host REDACTED:443: dial tcp: lookup REDACTED: no such host
```
```
# VPN connected
❯ time curl https://REDACTED/
<html><body>You are being <a href="https://REDACTED/sign_in">redirected</a>.</body></html>
real	0m0.221s
user	0m0.013s
sys	0m0.006s

# VPN disconnected
❯ time curl https://REDACTED/
curl: (56) CONNECT tunnel failed, response 502

real	0m0.041s
user	0m0.005s
sys	0m0.007s
```
```

--- a/pacfetcher.go
+++ b/pacfetcher.go
@@ -134,6 +134,12 @@ func (pf *pacFetcher) download() []byte {
 	}
 	pf.connected = false
 
+	// We've just detected a change in network state, so close any "idle"
+	// connections from the previous network. This forces a fresh DNS
+	// lookup and TCP dial during the next PAC download. For context, see
+	// <https://github.com/samuong/alpaca/issues/165>.
+	pf.client.CloseIdleConnections()
+
 	pacurl, err := pf.pacFinder.findPACURL()
 	if err != nil {
 		log.Printf("Error while trying to detect PAC URL: %v", err)

